BasicRPG/BasicRPG/Program.cs

```diff
@@ -1,160 +1,121 @@
         }
-        else if (characterChoice == 2)
-        {
-            Console.WriteLine($"From here on out you are {name} the cunning mage!");
-            player = new Warrior();
-        }
-
-        return player;
     }
-    static void battle(Character player, Character enemy)
+    static void Battle(Character player, Character enemy) 
     {
         Random random = new Random();
-        bool playerAlive = true;
-        bool enemyAlive = true;
-        int playerDamage;
-        int enemyDamage;
-        Console.WriteLine("You are now fighting the mighty GOBLIN!");
+        Console.WriteLine($"{player.name} is fighting {enemy.name}!");
 
-        while(playerAlive && enemyAlive)
+        while(player.IsAlive() && enemy.IsAlive()) 
         {
-            playerDamage = random.Next(1, 6);
-            enemyDamage = random.Next(1, 6);
             Console.WriteLine("Rolling the dice for player damage!");
             Console.ReadKey();
-            player.dealDamage(enemy, playerDamage);
+            player.DealDamage(enemy, random.Next(1, 6));
+            if (!enemy.IsAlive()) 
+            {
+                Console.WriteLine($"{enemy.name} has DIED! You are victorious!");
+                return;
+            }
             Console.WriteLine("Rolling the dice for enemy damage!");
             Console.ReadKey();
-            enemy.dealDamage(player, enemyDamage);
+            enemy.DealDamage(player, random.Next(1, 6));
+            if (!player.IsAlive()) 
+            {
+                Console.WriteLine($"{player.name} has DIED! You lost!");
+                return;
+            }
             Console.ReadKey();
-            playerAlive = player.isAlive();
-            enemyAlive = enemy.isAlive();
-        }
-
-        if (playerAlive == false)
-        {
-            Console.WriteLine("The player has DIED! You lost!");
-        }
-        else if (enemyAlive == false)
-        {
-            Console.WriteLine("You defeated the enemy! You are victorious!");
         }
-
-        Console.ReadKey();
-  
     }
-
 }
 
-public class Character
+public class Character 
 {
-    public bool alive;
-    public int playerHp = 20;
-    public int enemyHp = 20;
-    string name = " ";
-    public virtual void takeDamage(Character x)
-    {
+    int hp = 20;
+    public string name = "";
 
+    public Character(string name)
+    {
+        this.name = name;
     }
-    public virtual void dealDamage(Character x, int dmg)
+    public void DealDamage(Character defender, int damage) 
     {
+        defender.setHp(defender.hp - damage);
+        Console.WriteLine($"{name} has dealt {damage} damage to {defender.name}! {defender.name} has {defender.hp} hp left!");
+    }
 
+    public void setHp(int hp)
+    {
+        this.hp = hp;
     }
-    public virtual bool isAlive()
+    public bool IsAlive() 
     {
-        return playerHp > 0;  // The player is alive if playerHp > 0
+        return hp > 0; 
     }
```

Comment:
checking for hp is redundant when the parent `Character` class has a method that can be called to determine if it's alive or not.